data-products/src/main/scala/org/sunbird/analytics/sourcing/SourcingSummaryReport.scala

```diff
@@ -0,0 +1,124 @@
+  def main(config: String)(implicit sc: Option[SparkContext], fc: Option[FrameworkContext]): Unit = {
+    JobLogger.log(s"Started execution - $jobName", None, Level.INFO)
+    implicit val jobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    implicit val spark = openSparkSession(jobConfig)
+
+    try {
+      val res = CommonUtil.time(execute())
+      JobLogger.end(s"$jobName completed execution", "SUCCESS", Option(Map("timeTaken" -> res._1)))
+    } finally {
+      frameworkContext.closeContext()
+      spark.close()
+    }
+  }
+
+  // $COVERAGE-ON$ Enabling scoverage for all other functions
+  def execute()(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig) = {
+    val tables = JSONUtils.deserialize[Map[String, String]](JSONUtils.serialize(config.modelParams.get.getOrElse("tables", Map())))
+    val programData = loadData(url, tables("programTable")).withColumnRenamed("status", "programStatus")
+    val nominationData = loadData(url, tables("nominationTable"))
+    val projectDf = programData.join(nominationData, Seq("program_id"), "outer")
+      .select("program_id", "status", "rootorg_id", "user_id")
+
+    process(projectDf)
+  }
+
+  def process(projectDf: DataFrame)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig) = {
+    val userDf = getUserDetails()
+    val resultDf = userDf.join(projectDf, userDf.col("userId") === projectDf.col("user_id"), "outer")
+      .withColumnRenamed("user_id", "contributorId")
+      .withColumn("reportName", lit("SourcingSummaryReport"))
+      .withColumn("timestamp", lit(System.currentTimeMillis()))
+    JobLogger.log(s"resultDf count - ${resultDf.count()}", None, Level.INFO)
+
+    val modelParams = config.modelParams.get
+    val storageConfig = StorageConfig(AppConf.getConfig("cloud_storage_type"), "reports", "", Option(modelParams.getOrElse("storageKey", "druid_storage_account_key").toString), Option(modelParams.getOrElse("storageSecret", "druid_storage_account_secret").toString))
+    resultDf.saveToBlobStore(storageConfig, "json", "sourcing",
+      Option(Map("header" -> "true")), Option(List("reportName")))
+    submitReportToDruid(modelParams)
+  }
+
+  def loadData(url: String, tableName: String)(implicit spark: SparkSession, config: JobConfig): DataFrame = {
+    spark.read.jdbc(url, tableName, connProperties)
+  }
+
+  def submitReportToDruid(modelParams: Map[String, AnyRef]): Unit = {
+    JobLogger.log(s"Submitting Druid Ingestion Task", None, Level.INFO)
+    val segmentUrl = modelParams.getOrElse("druidSegmentUrl", "http://localhost:8081/druid/coordinator/v1/metadata/datasources/sourcing-model-snapshot/segments").asInstanceOf[String]
+    val deleteSegmentUrl = modelParams.getOrElse("deleteSegmentUrl", "http://localhost:8081/druid/coordinator/v1/datasources/sourcing-model-snapshot/segments/").asInstanceOf[String]
```

Comment:
Fetch only the host variable/baseurl from config. The API endpoint can be coded or can come from application.conf